test/foundry/module/SessionKeyManager/SessionKeyManagerHybrid.SingleCall.t.sol

```diff
@@ -0,0 +1,1069 @@
+        UserOperation memory op = makeEnableAndUseSessionUserOp(
+            getSmartAccountExecuteCalldata(
+                address(stub),
+                0,
+                abi.encodeCall(
+                    stub.emitMessage,
+                    ("shouldProcessTransactionFromSessionKey")
+                )
+            ),
+            sa,
+            0,
+            sessionKeyManagerHybrid,
+            sessionData,
+            bob,
+            0,
+            sessionEnableData,
+            sessionEnableSignature
+        );
+
+        vm.expectEmit();
+        emit SessionCreated(address(sa), sessionDataDigest, sessionData);
+        vm.expectEmit();
+        emit Log("shouldProcessTransactionFromSessionKey");
+        entryPoint.handleOps(toArray(op), owner.addr);
+
+        // Check session is enabled
+        ISessionKeyManagerModuleHybrid.SessionData
+            memory enabledSessionData = sessionKeyManagerHybrid
+                .enabledSessionsData(sessionDataDigest, address(sa));
+        assertEq(enabledSessionData, sessionData);
+    }
+
+    function testExplicitEnableAndUseSessionDifferentOp() public {
+        SessionKeyManagerHybrid.SessionData
+            memory sessionData = ISessionKeyManagerModuleHybrid.SessionData({
+                validUntil: 0,
+                validAfter: 0,
+                sessionValidationModule: address(mockSessionValidationModule),
+                sessionKeyData: abi.encodePacked(bob.addr)
+            });
+
+        // Generate Session Data
+        uint64[] memory chainIds = new uint64[](1);
+        chainIds[0] = uint64(block.chainid);
+
+        SessionKeyManagerHybrid.SessionData[]
+            memory sessionDatas = new SessionKeyManagerHybrid.SessionData[](1);
+        sessionDatas[0] = sessionData;
+
+        // Enable session
+        UserOperation memory op = makeEcdsaModuleUserOp(
+            getSmartAccountExecuteCalldata(
+                address(sessionKeyManagerHybrid),
+                0,
+                abi.encodeCall(
+                    sessionKeyManagerHybrid.enableSession,
+                    (sessionData)
+                )
+            ),
+            sa,
+            0,
+            alice
+        );
+        entryPoint.handleOps(toArray(op), owner.addr);
+
+        // Use session with just digest
+        op = makeUseExistingSessionUserOp(
+            getSmartAccountExecuteCalldata(
+                address(stub),
+                0,
+                abi.encodeCall(
+                    stub.emitMessage,
+                    ("shouldProcessTransactionFromSessionKey")
+                )
+            ),
+            sa,
+            0,
+            sessionKeyManagerHybrid,
+            sessionData,
+            bob
+        );
+        vm.expectEmit();
+        emit Log("shouldProcessTransactionFromSessionKey");
+        entryPoint.handleOps(toArray(op), owner.addr);
+    }
+
+    function testEnableAndUseSessionPostCaching() public {
```

Comment:
I find the names 'Cached' and 'Caching' here and further in the tests slightly confusing. However, it motivates to read the test and the code of the test is self-explanatory so it's not a big issue :)

Reply:
Updated the names to "SessionEnable"